core/src/main/java/org/polypheny/db/adapter/DataSource.java

```diff
@@ -23,20 +23,18 @@
 import lombok.AllArgsConstructor;
 import lombok.Getter;
 import org.pf4j.ExtensionPoint;
-import org.polypheny.db.catalog.entity.CatalogAdapter.AdapterType;
-import org.polypheny.db.catalog.entity.CatalogGraphDatabase;
-import org.polypheny.db.catalog.entity.CatalogGraphPlacement;
-import org.polypheny.db.prepare.Context;
+import org.polypheny.db.catalog.catalogs.StoreCatalog;
+import org.polypheny.db.catalog.entity.LogicalAdapter.AdapterType;
 import org.polypheny.db.type.PolyType;
 
-public abstract class DataSource extends Adapter implements ExtensionPoint {
+public abstract class DataSource<S extends StoreCatalog> extends Adapter<S> implements ExtensionPoint {
```

Comment:
Consider calling it `AdapterCatalog` or `PhysicalCatalog` instead of `StoreCatalog`.

Reply:
Already changed